aws/logs_monitoring/lambda_function.py

```diff
@@ -757,13 +786,22 @@ def normalize_events(events, metadata):
     for event in events:
         events_counter += 1
         if isinstance(event, dict):
-            normalized.append(merge_dicts(event, metadata))
+            normalized_event = merge_dicts(event, metadata)
         elif isinstance(event, str):
-            normalized.append(merge_dicts({"message": event}, metadata))
+            normalized_event = merge_dicts({"message": event}, metadata)
         else:
             # drop this log
             continue
 
+        # in case it's a cloudtrail event replace the hostname with the
+        # hostname in the ARN if available.
+        if normalized_event.get(DD_SOURCE) == "cloudtrail":
```

Comment:
It's better to keep `normalize_events` be a generic function. The best place to parse the message and override `DD_HOST` are within the log handler functions:

https://github.com/DataDog/datadog-serverless-functions/blob/615c4e7a43d65e542fa83ac5908d47f7601baab3/aws/logs_monitoring/lambda_function.py#L883 and https://github.com/DataDog/datadog-serverless-functions/blob/615c4e7a43d65e542fa83ac5908d47f7601baab3/aws/logs_monitoring/lambda_function.py#L805

For example, this is how we override RDS's host:
https://github.com/DataDog/datadog-serverless-functions/blob/615c4e7a43d65e542fa83ac5908d47f7601baab3/aws/logs_monitoring/lambda_function.py#L925

Reply:
The code was put after the normalization merging as the metadata overrides the DD_HOST of reach log, not just the one with the `arn` that we parse.
AFAIK the various other logs in `events` in `normalize_events` don't need to be from the same host

Reply:
I see, that makes sense since it depends on how aws writes the trails to s3 objects, i guess there is no guarantee the events in the same s3 object would be from the same host. 

But I would move these code to `enrich` where we already been doing something similar https://github.com/DataDog/datadog-serverless-functions/blob/3d061da905ffaa50acecce06f7f6f0c5dde3acf7/aws/logs_monitoring/lambda_function.py#L479-L488 For simplicity, I would also rename `extract_arn_hostname` to `extract_host_from_cloudtrails` (or something similar) and include all the checks, e.g., `if normalized_event.get(DD_SOURCE) == "cloudtrail":` all into that one single helper function.

```python
for event in events: 
    add_metadata_to_lambda_log(event) 
    extract_ddtags_from_message(event) 
    extract_host_from_cloudtrails(event)
```